src/main/java/bio/terra/service/common/gcs/GcsUriUtils.java

```diff
@@ -0,0 +1,49 @@
+package bio.terra.service.common.gcs;
+
+import com.google.cloud.storage.BlobId;
+
+public final class GcsUriUtils {
+  private static final String GS_BUCKET_PATTERN = "[a-z0-9_.\\-]{3,222}";
```

Comment:
Another minor comment, writing a pattern match this way is slower than it needs to be. Not sure how important performance is here, but the fix is pretty simple.
```suggestion
  private static final Pattern GS_BUCKET_PATTERN = Pattern.compile("[a-z0-9_.\\-]{3,222}");
  
  ...

    if (!GS_BUCKET_PATTERN.matcher(bucket).matches()) {    
  
```

It's more efficient this way because calling `String.matches()` calls `Pattern.compile()` every time it's called, and in this case the pattern is a constant.